Simplified/Reader2/NYPLReaderExtensions.swift

```diff
@@ -46,20 +46,25 @@ extension NYPLBook {
 
 @objc extension NYPLRootTabBarController {
   func presentBook(_ book: NYPLBook) {
-    guard let libModule = r2Owner?.libraryModule else {
+    guard let libraryService = r2Owner?.libraryService, let readerModule = r2Owner?.readerModule else {
       return
     }
 
-    let libService = libModule.libraryService
-
-    libService.parseBookAsync(book) { [weak self] publication in
+    libraryService.openBook(book, sender: self) { [weak self] result in
       guard let navVC = self?.selectedViewController as? UINavigationController else {
         preconditionFailure("No navigation controller, unable to present reader")
       }
-
-      self?.r2Owner.readerModule.presentPublication(publication,
-                                                    book: book,
-                                                    in: navVC) {}
+      switch result {
+      case .success(let publication):
+        readerModule.presentPublication(publication: publication, book: book, in: navVC) {
+          //
```

Comment:
is there a TODO: missing here or is it still ok to have an empty completion block?

Reply:
there was no TODO here before migration, it also seems we don't have anything to do here